### PR TITLE
[QF-4312] form check on change

### PR DESCRIPTION
**Pull request by afifvdin**
## Summary

Enhanced form validation and user experience in FormBuilder component with three key improvements:

1. **Real-time validation**: Changed form validation mode from `onBlur` to `onChange` to provide immediate validation feedback as users type
2. **Submit button state management**: Submit button is now automatically disabled when form has validation errors, preventing invalid form submissions and providing clear visual feedback
3. **Pristine form protection**: Submit button is disabled when form hasn't been modified yet (`isDirty` check), preventing unnecessary submissions of unchanged data

The submit button respects the `shouldSkipValidation` flag - if validation should be skipped, the button is only disabled during submission; otherwise, it's disabled when the form is pristine (unchanged), invalid, or submitting.

Closes: [QF-4312](https://quranfoundation.atlassian.net/browse/QF-4312)

## Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] ♻️ Refactoring (no functional changes)

## Test Plan

**Testing steps:**

1. Navigate to any page with forms using FormBuilder (e.g., Profile Settings, Change Password, Login/Signup)
2. Verify initial state:
   - Submit button should be disabled when form first loads (form is pristine)
3. Start filling in form fields with invalid data
4. Verify that:
   - Validation errors appear immediately as you type (onChange) rather than when you leave the field (onBlur)
   - Submit button remains disabled when form has validation errors
   - Submit button becomes enabled when all fields are valid AND form has been modified
   - Error messages clear immediately when input becomes valid
   - Submit button shows loading state during submission
5. Test edge case:
   - Make a valid change, verify button is enabled
   - Revert change back to original value, verify button is disabled again (form is no longer dirty)
6. Test with the following forms:
   - Change Password form (`/profile`)
   - Edit Details form (`/profile`)
   - Forgot Password form
   - Any collection/save forms
7. Verify forms with `shouldSkipValidation` flag still allow submission regardless of validation state (but button should still be disabled during submission)

**Forms affected:**
- Profile forms (Change Password, Edit Details)
- Authentication forms (Login, Forgot Password)
- Collection forms (Save to Collection)
- All other forms using FormBuilder component

### Code Quality

- [x] I have performed a **self-review** of my code (file by file)
- [x] My code follows the [project style guidelines](/.github/copilot-instructions.md)
- [x] No `any` types used (or justified if unavoidable)
- [x] No unused code, imports, or dead code included
- [x] Complex logic has inline comments explaining "why"
- [x] Functions are under 30 lines and follow single responsibility

### Testing & Validation

- [ ] All tests pass locally (`yarn test`)
- [x] Linting passes (`yarn lint`)
- [x] Build succeeds (`yarn build`)
- [x] Edge cases and error scenarios are handled

### Documentation

- [x] Code is self-documenting with clear naming
- [ ] README updated (if adding features or setup changes)
- [ ] Inline comments added for complex logic

### Localization (if UI changes)

- [ ] All user-facing text uses `next-translate`
- [ ] Only English locale files modified (Lokalise handles others)
- [ ] RTL layout verified

### Accessibility (if UI changes)

- [ ] Semantic HTML elements used
- [ ] ARIA attributes added where needed
- [ ] Keyboard navigation works

## AI Assistance Disclosure

- [x] AI tools were NOT used for this PR
- [ ] AI tools were used, and I have **thoroughly reviewed and validated** all generated code

[QF-4312]: https://quranfoundation.atlassian.net/browse/QF-4312?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/FormBuilder/FormBuilder.tsx [hunk 1]
```diff
@@ -57,7 +57,12 @@ const FormBuilder = <T,>({
   renderAction,
   shouldSkipValidation,
 }: FormBuilderProps<T>) => {
-  const { handleSubmit, control, setError } = useForm({ mode: 'onBlur' });
+  const {
+    handleSubmit,
+    control,
+    setError,
+    formState: { isValid },
+  } = useForm({ mode: 'onChange' });
```
[the remark this answers is not in the document]
Comment (afifvdin) [reply src/components/FormBuilder/FormBuilder.tsx:65]: we use option 3
